### PR TITLE
Add rz_lang_byte_array and remove rz_print_code

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Book issue here: https://github.com/rizinorg/book/issues/91

Extra tests have been added to cover all cases

## Removed
- `pcA  .bytes with instructions in comments` removes the dependency on RzCore
- `pci  C array of bytes with instructions` same reason as `pcA` and was returning same output as `pcA`
- `pcs  string` useless since `ps<cmd>` does the same and better.
- `pcP python` alias for `pcp`

## Moved
- `pcv  JaVa` is now `pcJ`
- `pcz  Swift` is now `pcs`
- `pcJ  javascript` is now `pcn` because the output can be only used on NodeJS.
- `pcS bash shellscript` is now `pcb`

## Added
- `pcg` for golang output

Also almost all the outputs (with the exception of ObjC and Vlang) has been tested and validated.

## New shell output

```
Usage: pc[?]   # Print bytes as code byte arrays.
| pc  # Generate a C/C++ byte array.
| pch # Generate a C/C++ 16 bits array.
| pcw # Generate a C/C++ 32 bits array.
| pcd # Generate a C/C++ 64 bits array.
| pca # Generate a byte array in GAS assembly.
| pcb # Generate a bash script with the byte array.
| pcg # Generate a Golang byte array.
| pcJ # Generate a Java byte array.
| pcj # Generate a JSON byte array.
| pck # Generate a Kotlin byte array.
| pcn # Generate a NodeJS buffer.
| pco # Generate a Objective-C/C++ byte array.
| pcp # Generate a Python byte array.
| pcr # Generate a Rust byte array.
| pcs # Generate a Swift byte array.
| pcy # Generate a Yara match pattern.
| pc* # Generate a rizin commands for writing the byte array.

Useful modifiers:
| pch @e:cfg.bigendian=<true|false> # Change endianness for pch, pcw and pcd commands
| pc @! <n>                     # Change the N of bytes (i.e. block size).

Example of usages:
| pch @! 64 @e:cfg.bigendian=true # Generate a C 32 bits array in big endian format, using 64 bytes
| pcp @! 1024                   # Generate a Python byte array of size 1024
| pcj @! 10                     # Generate a JSON bytes array of size 10
```

### CODE REVIEW

File: librz/core/cmd/cmd_print.c [hunk 1]
```diff
@@ -6124,7 +6124,70 @@ RZ_IPI int rz_cmd_print(void *data, const char *input) {
 				}
 				rz_cons_printf(".equ shellcode_len, %d\n", len);
 			} else {
-				rz_print_code(core->print, core->offset, core->block, len, input[1]);
+				char *str = NULL;
+				bool big_endian = rz_config_get_b(core->config, "cfg.bigendian");
+				// rz_print_code(core->print, core->offset, core->block, len, input[1]);
```
Comment (XVilka) [inline librz/core/cmd/cmd_print.c]: Remove this?

File: librz/core/cmd/cmd_print.c [hunk 2]
```diff
@@ -6124,7 +6124,70 @@ RZ_IPI int rz_cmd_print(void *data, const char *input) {
...
+					break;
+				case 'P': // "pcP" // python
+				case 'p': // "pcp" // python
+					str = rz_lang_byte_array(core->block, len < 0 ? 0 : len, RZ_LANG_BYTE_ARRAY_PYTHON);
+					break;
+				case 'h': // "pch" // C half words with asm
+					str = rz_lang_byte_array(core->block, len < 0 ? 0 : len, big_endian ? RZ_LANG_BYTE_ARRAY_C_CPP_HALFWORDS_BE : RZ_LANG_BYTE_ARRAY_C_CPP_HALFWORDS_LE);
+					break;
+				case 'w': // "pcw" // C words with asm
+					str = rz_lang_byte_array(core->block, len < 0 ? 0 : len, big_endian ? RZ_LANG_BYTE_ARRAY_C_CPP_WORDS_BE : RZ_LANG_BYTE_ARRAY_C_CPP_WORDS_LE);
```
Comment (XVilka) [inline librz/core/cmd/cmd_print.c:6173]: split the line?

File: librz/util/lang_byte_array.c [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+	rz_strbuf_append(sb, "let byteArray : [UInt8] = [\n ");
+	for (size_t pos = 0; pos < size; pos++) {
+		if (pos > 0 && !(pos % (RZ_LANG_BYTE_ARRAY_TRUNK_SIZE))) {
+			rz_strbuf_append(sb, "\n ");
+		}
+		rz_strbuf_appendf(sb, (pos + 1) < size ? " 0x%02x," : " 0x%02x\n];", buffer[pos]);
+	}
+}
+
+static void lang_byte_array_vlang(RzStrBuf *sb, const ut8 *buffer, size_t size) {
```
Comment (XVilka) [inline librz/util/lang_byte_array.c]: I suggest removing V lang support completely. It's too niche, at least for now.

**Review by XVilka — APPROVED** [review]
LGTM apart from a couple nitpicks.

File: librz/core/cmd/cmd_print.c (hunk 1 above)
Comment (wargio) [reply librz/core/cmd/cmd_print.c]: absolutely

File: librz/core/cmd/cmd_print.c (hunk 2 above)
Comment (wargio) [reply librz/core/cmd/cmd_print.c:6173]: tbh, that is old shell. i would not care too much

File: librz/util/lang_byte_array.c (hunk 3 above)
Comment (wargio) [reply librz/util/lang_byte_array.c]: i will wait for the others comments.

**Review by XVilka — APPROVED** [review]

**Review by ret2libc — COMMENTED** [review]
I would remove all the `n_bytes` arguments as they just provide a way to replace the blocksize, which can already be done with `@!`.

Niiice anyway!

**Comment by wargio** [comment]
> I would remove all the `n_bytes` arguments as they just provide a way to replace the blocksize, which can already be done with `@!`.
> 
> Niiice anyway!

i had the same thought, but i wanted to double check with you

**Review by XVilka — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]

**Pull request merged**